### DISCUSSION TITLE
Nostr integration

### DISCUSSION THREAD

**Discussion by outlaw-dame** [assemblee-virtuelle/activitypods#199] [General]
I know we have discussed integration of Granary as middleware for multi-protocol translation largely in the context of supporting Atproto. After doing some research it may be easier to integrate Nostr as it’s a relatively simple protocol that focuses on the client and relays. It seems ideal for an initial test to handle another social protocol https://nostr.com/  https://github.com/nostr-protocol/nostr

**Comment by melvincarvalho** [comment]
Hi there.  2nd contributor to nostr here.  I'm also doing some integrations with Solid and nostr which is powered by schnorr signatures

- vocab, this could be a one-liner in the profile (pubkey + literal) : https://w3id.org/nostr
- In SolidOS chat we are already using schnorr signatures and key pairs : https://github.com/SolidOS/solid-ui/blob/66447a47fbefbec623b4c8ba09a7988817dc2248/src/chat/keys.ts#L2
- I am working on an authentication method for solid+nostr : https://nostrcg.github.io/http-schnorr-auth/

Demo of login: https://melvin.solid.social/public/login1.mp4

Demo of login with extension: https://melvin.solid.social/public/schnorr.mp4

**Comment by niko-ng** [comment]
Hello. One question I already raised on other forums: why write a new spec called Schnorr Signatures when there is already RFC9421 that is the standard everybody is using? Also, this spec "http-schnorr-auth" seems badly written. In the spec, we read, towards the end, "The client includes the base64-encoded event in the Authorization header with the scheme Schnorr" but then examples are given where "Nostr" is the scheme. This just looks like not very serious draft, in my opinion. If Nostr wants to facilitate integration with others, maybe it should start by using standards and not reinvent the wheel.

**Reply by niko-ng** [reply]
when i wrote "everybody is using" i meant "RFC9421 or its ancestor cavage", sorry for the confusion.
